homeassistant/components/fortios/device_tracker.py

```diff
@@ -22,101 +21,134 @@
 from homeassistant.helpers.typing import ConfigType
 
 _LOGGER = logging.getLogger(__name__)
-DEFAULT_VERIFY_SSL = False
 
-
-PLATFORM_SCHEMA = BASE_PLATFORM_SCHEMA.extend(
+PLATFORM_SCHEMA = PARENT_PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_HOST): cv.string,
         vol.Required(CONF_TOKEN): cv.string,
-        vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): cv.boolean,
+        vol.Optional(CONF_VERIFY_SSL, default=False): cv.boolean,
     }
 )
 
 
 def get_scanner(hass: HomeAssistant, config: ConfigType) -> FortiOSDeviceScanner | None:
-    """Validate the configuration and return a FortiOSDeviceScanner."""
-    host = config[DOMAIN][CONF_HOST]
-    verify_ssl = config[DOMAIN][CONF_VERIFY_SSL]
-    token = config[DOMAIN][CONF_TOKEN]
-
-    fgt = FortiOSAPI()
-
-    try:
-        fgt.tokenlogin(host, token, verify_ssl)
-    except ConnectionError as ex:
-        _LOGGER.error("ConnectionError to FortiOS API: %s", ex)
-        return None
-    except Exception as ex:  # pylint: disable=broad-except
-        _LOGGER.error("Failed to login to FortiOS API: %s", ex)
-        return None
-
-    status_json = fgt.monitor("system/status", "")
-
-    current_version = AwesomeVersion(status_json["version"])
-    minimum_version = AwesomeVersion("6.4.3")
-    if current_version < minimum_version:
-        _LOGGER.error(
-            "Unsupported FortiOS version: %s. Version %s and newer are supported",
-            current_version,
-            minimum_version,
-        )
-        return None
+    """Validate the configuration and return a FortiOS scanner."""
+    scanner = FortiOSDeviceScanner(config[DOMAIN])
 
-    return FortiOSDeviceScanner(fgt)
+    return scanner if scanner.initialize() else None
 
 
 class FortiOSDeviceScanner(DeviceScanner):
     """Class which queries a FortiOS unit for connected devices."""
 
-    def __init__(self, fgt) -> None:
+    def __init__(self, config):
         """Initialize the scanner."""
-        self._clients: list[str] = []
-        self._clients_json: dict[str, Any] = {}
-        self._fgt = fgt
+        self.host = config[CONF_HOST]
+        self.token = config[CONF_TOKEN]
+        self.verify_ssl = config[CONF_VERIFY_SSL]
+        self.last_results = {}
+        self._fgt = None
 
-    def update(self):
-        """Update clients from the device."""
-        clients_json = self._fgt.monitor("user/device/query", "")
-        self._clients_json = clients_json
+    def initialize(self) -> bool:
```

Comment:
Don't add this method. It's a bad pattern. The api instance should not be created after we we create the scanner instance.

Reply:
OK - I can remove it. It was suggested by your team, but no worries. I'll remove all refactoring only apply a fix for the open issue.

Reply:
refactoring removed and added fix to the open issue